### ISSUE TITLE
Refactor FDE section

### ISSUE THREAD

**Issue by flexibeast** [void-linux/void-docs#284]

**Comment by jeffayle** [comment]
I think only three sections need to be kept:
- prepare the disks
- configure dracut
- configure grub

This outline will be the same for a number of  root-filesystem-is-on-whatever guides, so it might make sense to have a general overview, and then have more specific pages for things like full disk encryption, zfs, raid or whatever else that tell you the particular commands you need to use.

Or, it might make more sense to have pages for advanced disk configuration, advanced dracut config and advanced grub config, each of which would cover that section for a number of different setups (probably just in the format of use these commands to setup lvm/luks/raid/whatever with links to their man pages)

That's further down the road though, when the zfs guide gets refactored and merged, we'll have two similar guides to compare and decide what kind of abstraction makes sense.

**Comment by ericonr** [comment]
So the types of guides we can have are:

- One guide for each type of setup, like FDE and ZFS, which points out a part of the chroot guide to follow and then some specific steps to follow,
- One guide for each section of the chroot guide, with the chroot guide including guidance like "to customize this step, look at []"

I'm in favour of the last one, I think, and I'd prefer to refactor FDE to fit into the definitive style.

**Comment by jeffayle** [comment]
I'm in favour of the last one as well. The only issue I can foresee is that some setups (like full disk encryption) require you to do something, then go back to the original guide and come back to do something else later. It won't always be obvious to readers at a future step that they need to look into the advanced section of that step to complete something they'd started earlier.

**Comment by ericonr** [comment]
> It won't always be obvious to readers at a future step that they need to look into the advanced section of that step to complete something they'd started earlier.

Yeah, this might make them too fragmented... I might try to write in the two styles and see how it goes. Btw, @abenson do you have any suggestions for this?

**Comment by jeffayle** [comment]
If it helps, here's a draft of instructions for arm (both platformfs and rootfs/xbps) written in style (1): https://gist.github.com/jeffayle/d09d8baaa4b036a01d69ac63bec279f2

I'm not very happy with it and part of it is just being unedited, but it's hard to write instructions in that way. I can't think of any other way than  "follow the instructions until this step, and then do this instead, then go back to the instructions until this other step, and then do this instead".

**Comment by ericonr** [comment]
I don't think we need to go into as much detail as Arch or Gentoo. If you're interested in trying more advanced stuff, most of it is transferrable to Void, we mostly need to add the Void specific bits and instructions for a basic install.

I also like your initial guide. I feel like we should attempt to link to something for the "platform documentation" or even have a table for the platforms Void officially supports. You didn't mention proot, right? And I figured a RPi 3 could boot from a RPi 1 image, tbh. Can you check if it can chroot into one? Then it's just applications that are armv6l, not the kernel.

**Comment by jeffayle** [comment]
Yeah, linking to documentation for each platform we have binaries for is a good idea. I think that's where most of the arm documentation should come from. I can chroot into an armv6l installation on my rpi 3. I suspect the incompatibility is with the SoC or boot rom, not the processor ISA.
